lib/elixir_analyzer/exercise_test/common_checks/unless_with_else.ex

```diff
@@ -0,0 +1,72 @@
+defmodule ElixirAnalyzer.ExerciseTest.CommonChecks.UnlessWithElse do
+  @moduledoc """
+  Reports when an unless clause is used followed by an else clause.
+
+  Common check to be run on every single solution.
+  """
+  alias ElixirAnalyzer.Comment
+  alias ElixirAnalyzer.Constants
+
+  @spec run(Macro.t()) :: [{:pass | :fail | :skip, %Comment{}}]
+  def run(ast) do
+    {_ast, else_block} = Macro.prewalk(ast, [], &traverse/2)
+
+    if Enum.empty?(else_block) do
+      []
+    else
+      [
+        {:fail,
+         %Comment{
+           type: :actionable,
+           comment: Constants.solution_unless_with_else()
+         }}
+      ]
+    end
+  end
+
+  defp traverse({:@, _, [{:unless, _, _}]} = ast, issues) do
+    if else_block?(ast) do
+      else_block = else_block_for(ast)
+      {ast, [else_block | issues]}
+    else
+      {ast, []}
+    end
+  end
+
+  defp traverse({:unless, _, _} = ast, issues) do
```

Comment:
You are getting errors because you don't provide a default option for nodes that don't match either  `{:unless, _, _} ` or `{:@, _, [{:unless, _, _}]}` (which BTW I don't think you need since that would be matching code like `@unless x`). You need to add
```elixir
defp traverse(node, issues), do: node
```
Or something similar. Make sure you check code like
```elixir
quote do
  unless x, do: :a, else: :b
end
```
in `iex` to look at the structure of how `unless` works.